openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -357,9 +357,19 @@ def starts_within(self, days):
         """
         Returns True if the course starts with-in given number of days otherwise returns False.
         """
-
         return course_metadata_utils.course_starts_within(self.start, days)
 
+    def retrieve_datetime(self, get_start=True):
```

Comment:
Please rename and change the signature of this method so there are separate methods for each date property of the course: `start_datetime` and `end_datetime`.  This makes it easier to understand the code that calls `retrieve_datetime`.  Cause it's unclear from the current name that it returns the start-date.